hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java

```diff
@@ -205,47 +241,51 @@ private void resolveMutableBuilderAttributes(
             } else {
                 builder.submitKey(newSubmitKey);
             }
-        } else {
-            builder.submitKey(topic.submitKey());
         }
         if (op.hasMemo()) {
             builder.memo(op.memo());
-        } else {
-            builder.memo(topic.memo());
         }
-        final var resolvedExpiryMeta = resolvedUpdateMetaFrom(handleContext.expiryValidator(), op, topic);
+        final var resolvedExpiryMeta = resolvedUpdateMetaFrom(handleContext.expiryValidator(), op, currentExpiryMeta);
         builder.expirationSecond(resolvedExpiryMeta.expiry());
         builder.autoRenewPeriod(resolvedExpiryMeta.autoRenewPeriod());
         if (op.hasAutoRenewAccount() && designatesAccountRemoval(op.autoRenewAccount())) {
             builder.autoRenewAccountId((AccountID) null);
         } else {
             builder.autoRenewAccountId(resolvedExpiryMeta.autoRenewAccountId());
         }
+        if (op.hasFeeScheduleKey()) {
+            final var newFeeScheduleKey = op.feeScheduleKey();
+            if (newFeeScheduleKey == null || isKeyRemoval(newFeeScheduleKey)) {
```

Comment:
This can't be null here

Reply:
Done. Even though Intelij doesn't like it and throws warning that NPE may occur.